### PR TITLE
CLOUD-412 Add auto-tuning test to Jenkinsfile

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| self-healing | passed |
| scaling | passed |
| init-deploy | failed |
| scaling-proxysql | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| auto-tuning | passed |
| scheduled-backup | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| recreate | passed |
| self-healing | passed |
| scaling | failed |
| operator-self-healing | failed |
| demand-backup | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | failed |
| scaling | failed |
| init-deploy | failed |
| recreate | passed |
| demand-backup | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| self-healing | failed |
| recreate | passed |
| scaling | failed |
| demand-backup | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| recreate | passed |
| init-deploy | failed |
| scaling | passed |
| scaling-proxysql | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| auto-tuning | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| scaling | passed |
| init-deploy | failed |
| self-healing | failed |
| scaling-proxysql | passed |
| demand-backup | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| recreate | passed |
| self-healing | failed |
| scaling | passed |
| scaling-proxysql | passed |
| demand-backup | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| scaling | passed |
| self-healing | failed |
| init-deploy | failed |
| scaling-proxysql | passed |
| demand-backup | passed |
| scheduled-backup | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | failed |
| recreate | passed |
| scaling | passed |
| demand-backup | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| scheduled-backup | passed |
| storage | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | failed |
| init-deploy | failed |
| recreate | passed |
| scaling | passed |
| scaling-proxysql | passed |
| demand-backup | passed |
| scheduled-backup | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling | passed |
| scaling-proxysql | passed |
| storage | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| limits | failed |
| auto-tuning | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| init-deploy | passed |
| self-healing | passed |
| recreate | passed |
| scaling-proxysql | passed |
| storage | failed |
| operator-self-healing | passed |
| demand-backup | passed |
| auto-tuning | passed |
| scheduled-backup | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| recreate | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| limits | passed |
| monitoring | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| self-healing | passed |
| scaling | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| storage | passed |
| limits | passed |
| auto-tuning | passed |
| monitoring | passed |
| scheduled-backup | failed |
| monitoring-2-0 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| scaling | passed |
| init-deploy | passed |
| recreate | passed |
| scaling-proxysql | passed |
| storage | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| limits | passed |
| auto-tuning | passed |
| monitoring | passed |
| monitoring-2-0 | failed |
| scheduled-backup | failed |

File: e2e-tests/auto-tuning/run [hunk 1]
```diff
@@ -7,29 +7,61 @@ test_dir=$(realpath $(dirname $0))
...
+    echo "max_connections is set to ${AUTO_CONNECTIONS}, which does not correlate with cr.pxc.limits.memory / ${RDS_MEM_INSTANCE}"
+    exit 1
+fi
+
+kubectl_bin delete pxc --all
+kubectl_bin delete pvc --all
+
+sleep 10
+
+spinup_pxc "$cluster" "${test_dir}/conf/${cluster}2.yml"
```
Comment (delgod) [inline e2e-tests/auto-tuning/run]: can we just apply "${test_dir}/conf/${cluster}2.yml" without deleting cluster?

File: e2e-tests/auto-tuning/run [hunk 2]
```diff
@@ -7,29 +7,61 @@ test_dir=$(realpath $(dirname $0))
...
 
+sleep 10
 
-wait_cluster_consistency "${cluster}" 3
-sleep 20
-wait_cluster_consistency "${cluster}" 3
-sleep 20
-wait_cluster_consistency "${cluster}" 3
-sleep 20
+spinup_pxc "$cluster" "${test_dir}/conf/${cluster}3.yml"
```
Comment (delgod) [inline e2e-tests/auto-tuning/run]: can we just apply "${test_dir}/conf/${cluster}3.yml" without deleting cluster?

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| init-deploy | passed |
| self-healing | passed |
| recreate | passed |
| scaling-proxysql | passed |
| storage | passed |
| operator-self-healing | passed |
| limits | passed |
| monitoring | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| auto-tuning | passed |
| affinity | failed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| self-healing | passed |
| scaling | passed |
| init-deploy | failed |
| scaling-proxysql | failed |
| operator-self-healing | passed |
| demand-backup | passed |
| auto-tuning | failed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| scaling | passed |
| self-healing | passed |
| scaling-proxysql | failed |
| operator-self-healing | passed |
| init-deploy | passed |
| auto-tuning | passed |
| storage | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| recreate | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| init-deploy | passed |
| storage | failed |
| scaling | passed |
| scaling-proxysql | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| demand-backup | passed |
| scaling-proxysql | passed |
| scheduled-backup | failed |
| init-deploy | passed |
| storage | failed |
| self-healing | passed |
| operator-self-healing | passed |
| auto-tuning | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| scaling | passed |
| demand-backup | passed |
| scaling-proxysql | passed |
| scheduled-backup | failed |
| self-healing | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| auto-tuning | passed |
| storage | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |

File: e2e-tests/auto-tuning/run (hunk 2 above)
Comment (cap1984) [reply e2e-tests/auto-tuning/run]: done

File: e2e-tests/auto-tuning/run (hunk 1 above)
Comment (cap1984) [reply e2e-tests/auto-tuning/run]: done

File: e2e-tests/auto-tuning/conf/auto-tuning2.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-apiVersion: pxc.percona.com/v1
+apiVersion: pxc.percona.com/v1-3-0
```
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/auto-tuning/conf/auto-tuning2.yml]: done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| scaling | passed |
| self-healing | passed |
| init-deploy | passed |
| scaling-proxysql | failed |
| storage | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| limits | passed |
| auto-tuning | passed |
| monitoring | passed |
| scheduled-backup | failed |
| monitoring-2-0 | passed |
| affinity | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| recreate | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| auto-tuning | passed |
| scheduled-backup | failed |
| init-deploy | passed |
| storage | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |

**Pull request merged**